.github/workflows/build_publish_builder_image.yaml

```diff
@@ -0,0 +1,73 @@
+name: Build and Push Builder Image
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      go-version:
+        description: 'Go version'
+        required: false
+        default: ''
+      ubuntu-version:
+        description: 'Ubuntu version'
+        required: false
+        default: ''
+
+jobs:
+  build-and-push:
+    name: Build and Push Builder Image
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go-version: [1.22.9, 1.23.2]
+        ubuntu-version: [16.04]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set version
+        id: set_version
+        run: |
+          VERSION="${{ github.run_number }}-$(date +'%Y%m%d%H%M%S')"
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+      
+      - name: Set Go and Ubuntu versions
+        id: set_versions
+        run: |
+          if [ "${{ github.event.inputs.go-version }}" != "" ]; then
+            echo "GO_VERSION=${{ github.event.inputs.go-version }}" >> $GITHUB_ENV
+          else
+            echo "GO_VERSION=${{ matrix.go-version }}" >> $GITHUB_ENV
+          fi
+          if [ "${{ github.event.inputs.ubuntu-version }}" != "" ]; then
+            echo "UBUNTU_VERSION=${{ github.event.inputs.ubuntu-version }}" >> $GITHUB_ENV
+          else
+            echo "UBUNTU_VERSION=${{ matrix.ubuntu-version }}" >> $GITHUB_ENV
+          fi
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: build/Dockerfile
+          build-args: |
+            GO_VERSION=${{ matrix.go-version }}
+            UBUNTU_VERSION=${{ matrix.ubuntu-version }}
+          push: true
+          tags: |
+            ghcr.io/${{ github.repository_owner }}/${{ github.event.repository.name }}:${{ env.VERSION }}-go${{ matrix.go-version }}-ubuntu${{ matrix.ubuntu-version }}
```

Comment:
is someone using this or it is useful just for keeping track of releases?

Reply:
It will be used by all the OHIs as the base builder image for compiling their binaries. One example [here](https://github.com/newrelic/nri-docker/pull/264)

Reply:
yes but we use latest, not the one having "RUN_AND_TIME"

Reply:
Same rationale as https://github.com/newrelic/coreint-automation/pull/89#discussion_r1850536078

Reply:
Simplified, only using `latest` now.